includes/class-wc-payments-incentives-service.php

```diff
@@ -0,0 +1,208 @@
+<?php
+/**
+ * Class WC_Payments_Incentives_Service
+ *
+ * @package WooCommerce\Payments
+ */
+
+if ( ! defined( 'ABSPATH' ) ) {
+	exit; // Exit if accessed directly.
+}
+
+use WCPay\Database_Cache;
+
+/**
+ * Class handling onboarding related business logic.
+ */
+class WC_Payments_Incentives_Service {
+
+	/**
+	 * Cache util for managing onboarding data.
+	 *
+	 * @var Database_Cache
+	 */
+	private $database_cache;
+
+	/**
+	 * Class constructor
+	 *
+	 * @param Database_Cache $database_cache      Database cache util.
+	 */
+	public function __construct( Database_Cache $database_cache ) {
+		$this->database_cache = $database_cache;
+
+		add_action( 'wcpay_js_settings', [ $this, 'add_js_settings' ] );
```

Comment:
This should be `add_filter`. It works because `add_action` is actually `add_filter` under the hood, but it's best to keep patterns for readability and clarity of intent.

Reply:
💯 Not sure why I used `add_action` here, probably was just a copy/paste mistake. Good catch! Although as we mentioned in Slack, we'll not add the filter in this PR.